### ISSUE TITLE
Integrate cargo-careful

### ISSUE THREAD

**Issue by elichai** [rust-fuzz/cargo-fuzz#331]
cargo-careful enables debug assertions and some UB checks in libstd, I think it would be useful if we can fuzz under that configuration
https://github.com/RalfJung/cargo-careful

(@RalfJung Might be of interest to you)

**Comment by elichai** [comment]
I see now there's an open PR to enable `-Zbuild-std` https://github.com/rust-fuzz/cargo-fuzz/pull/292

**Comment by fitzgen** [comment]
Does this have any advantages over adding `-Zbuild-std` support given that we already have sanitizer support?

**Comment by elichai** [comment]
@fitzgen `-Zbuild-std` together with `-Cdebug-assertions=on -Zextra-const-ub-checks -Zstrict-init-checks --cfg "careful"` will basically give `cargo-careful`, the upside of these over just build-std+sanitizer is:
1. Add more UB checks in const evaluation (these never reach LLVM).
3. Add assertions that could catch invariants failing even if they haven't manifested into UB just yet (I think), and catch rust-specific UB that aren't necessarily LLVM UB.

**Comment by fitzgen** [comment]
Yeah I guess I would rather use these `-Z` rustc flags ourselves than add *another* layer around `cargo`.
